src/core/src/op/mod.cpp

```diff
@@ -31,6 +48,182 @@ struct Evaluate : ov::element::NoAction<bool> {
+                *out_first++ =
+                    reference::func::mod_interval(*v_lb_first++, *v_ub_first++, *m_lb_first++, *m_ub_first++).second;
+            }
+        }
+        return true;
+    }
+};
+
+namespace {
+
+/**
+ * @brief Get node inputs bounds as TensorVector.
+ *
+ * The inputs bounds are stored as [lower0, upper0, lower1, upper1].
+ *
+ * @param op  Pointer to the node.
+ * @return Vector with inputs bounds tensors.
+ */
+TensorVector get_bounds(const Node* const op) {
+    auto&& v_bounds = ov::evaluate_both_bounds(op->input_value(0));
+    auto&& m_bounds = ov::evaluate_both_bounds(op->input_value(1));
+    return {std::move(v_bounds.first),
+            std::move(v_bounds.second),
+            std::move(m_bounds.first),
+            std::move(m_bounds.second)};
+}
+
+/**
+ * @brief Check if all bounds in vector are valid (allocated).
+ *
+ * @param bounds  TensorVector of bounds for check.
+ * @return True if bounds area valid otherwise false.
+ */
+bool are_bounds_valid(const TensorVector& bounds) {
+    return std::all_of(bounds.begin(), bounds.end(), [](const Tensor& t) {
+        return static_cast<bool>(t);
+    });
+}
+
+/**
+ * @brief Evaluate binary mask of values which cannot be calculated by modulo.
+ *
+ * @param output_mask  Output tensor to store result.
+ * @param bounds       Modulo inputs bounds.
+ * @return True if output_mask has got valid data otherwise false.
+ */
+bool evaluate_undefined_result_mask(Tensor& output_mask, const TensorVector& bounds) {
+    const auto eq_op = v1::Equal();
+    const auto or_op = v1::LogicalOr();
+
+    const auto& in_et = bounds.front().get_element_type();
+
+    auto zero_t = ov::util::make_tensor_of_value(in_et, 0);
+    auto max_t = ov::util::make_tensor_of_max_value(in_et);
+
+    const auto& v_ub = bounds[1];
+    const auto& m_lb = bounds[2];
+    const auto& m_ub = bounds[3];
+
+    auto output_mask_vector = TensorVector{output_mask};
+
+    if (!eq_op.evaluate(output_mask_vector, {m_lb, zero_t})) {
+        return false;
+    }
+
+    auto tmp_mask = TensorVector{{element::boolean, output_mask.get_shape()}};
+    if (!eq_op.evaluate(tmp_mask, {m_ub, zero_t})) {
+        return false;
+    }
+    or_op.evaluate(output_mask_vector, {output_mask_vector.front(), tmp_mask.front()});
+
+    if (!eq_op.evaluate(tmp_mask, {m_lb, max_t})) {
+        return false;
+    }
+    or_op.evaluate(output_mask_vector, {output_mask_vector.front(), tmp_mask.front()});
+
+    if (!eq_op.evaluate(tmp_mask, {v_ub, max_t})) {
+        return false;
+    }
+    return or_op.evaluate(output_mask_vector, {output_mask_vector.front(), tmp_mask.front()});
+}
+
+/**
+ * @brief Get the m (modulo) inputs bound with set to one values which can cause undefined result.
+ *
+ * @param bounds  Modulo operator inputs bounds.
+ * @param mask    Mask with undefined result values.
+ * @return Vector of bounds tensors.
+ */
+TensorVector get_m_bounds(const TensorVector& bounds, const Tensor& mask) {
```

Comment:
The bounds are same tensors just inserted into vectors (there is no data copy). 
The select operator replace data in existing tensor but it depends on `select` operator implementation how data are replaced.

I think most operators do like copy/transform input to output.
Is there any op which can apply changes on input tensor?

Reply:
No it cannot be changed. 
The input bounds are currently bounds of `Mod` inputs (others nodes). If this tensors will be modified it will impact on bound calculation in other steps.